src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.PropertySetDataFlowOperationVisitor.cs

```diff
@@ -0,0 +1,198 @@
+
+            protected override PropertySetAbstractValue GetAbstractValue(AbstractLocation location)
+                => this.CurrentAnalysisData.TryGetValue(location, out var value) ? value : ValueDomain.Bottom;
+
+            protected override void ResetCurrentAnalysisData() => ResetAnalysisData(CurrentAnalysisData);
+
+            protected override void StopTrackingAbstractValue(AbstractLocation location) => CurrentAnalysisData.Remove(location);
+
+            protected override void SetAbstractValue(AbstractLocation location, PropertySetAbstractValue value)
+            {
+                if (value != PropertySetAbstractValue.Unknown
+                    || this.CurrentAnalysisData.ContainsKey(location))
+                {
+                    this.CurrentAnalysisData[location] = value;
+                }
+            }
+
+
+            protected override PropertySetAnalysisData MergeAnalysisData(PropertySetAnalysisData value1, PropertySetAnalysisData value2)
+                => BinaryFormatterAnalysisDomainInstance.Merge(value1, value2);
+            protected override PropertySetAnalysisData GetClonedAnalysisData(PropertySetAnalysisData analysisData)
+                => GetClonedAnalysisDataHelper(analysisData);
+            protected override PropertySetAnalysisData GetEmptyAnalysisData()
+                => GetEmptyAnalysisDataHelper();
+            protected override PropertySetAnalysisData GetAnalysisDataAtBlockEnd(PropertySetAnalysisResult analysisResult, BasicBlock block)
+                => GetClonedAnalysisDataHelper(analysisResult[block].OutputData);
+            protected override bool Equals(PropertySetAnalysisData value1, PropertySetAnalysisData value2)
+                => EqualsHelper(value1, value2);
+
+            protected override void SetValueForParameterPointsToLocationOnEntry(IParameterSymbol parameter, PointsToAbstractValue pointsToAbstractValue)
+            {
+            }
+
+            protected override void EscapeValueForParameterPointsToLocationOnExit(IParameterSymbol parameter, AnalysisEntity analysisEntity, PointsToAbstractValue pointsToAbstractValue)
+            {
+            }
+
+            protected override void SetAbstractValueForArrayElementInitializer(IArrayCreationOperation arrayCreation, ImmutableArray<AbstractIndex> indices, ITypeSymbol elementType, IOperation initializer, PropertySetAbstractValue value)
+            {
+            }
+
+            protected override void SetAbstractValueForAssignment(IOperation target, IOperation assignedValueOperation, PropertySetAbstractValue assignedValue, bool mayBeAssignment = false)
+            {
+            }
+
+            public override PropertySetAbstractValue VisitObjectCreation(IObjectCreationOperation operation, object argument)
+            {
+                PropertySetAbstractValue abstractValue = base.VisitObjectCreation(operation, argument);
+                if (operation.Type != null && operation.Type == this.DeserializerTypeSymbol)
```

Comment:
>operation.Type != null [](start = 20, length = 22)

Redundant check. Object creation operation is guaranteed to have a non-null type.